src/main/kotlin/pl/edu/uj/ii/ksi/mordor/services/repository/RepositoryService.kt

```diff
@@ -11,19 +12,15 @@ import java.nio.file.Paths
 @Service
 class RepositoryService(@Value("\${mordor.root_path}") private val rootPathStr: String) {
     private final val rootPath: Path = Paths.get(rootPathStr)
+    private final val logger = LoggerFactory.getLogger(RepositoryService::class.java)
 
     private fun getRequestedPath(path: String): Path {
-        val parts = path.split("/")
-        var current = rootPath
-        for (i in 2..(parts.size - 1)) {
-            val part = parts[i]
-            when {
-                part == "." || part == ".." -> throw BadRequestException("invalid path")
-                part != "" -> current = current.resolve(part)
-            }
-
+        val fullPath = rootPath.resolve(path).normalize().toAbsolutePath()
+        if (!fullPath.startsWith(rootPath)) {
+            logger.warn("Tried to access file outside of repository root: $path")
```

Comment:
Do we really need to log this? If so, can it be reduced to info or debug level?

Reply:
@apardyl In my opinion it is wise to log everything that might be useful. Accessing file outside of repository root is suspicious enough to be logged. In this case, I agree that it might be better to reduce to debug level.